activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -619,12 +619,9 @@ def default_index_type?(index) # :nodoc:
       end
 
       def build_insert_sql(insert) # :nodoc:
-        sql = +"INSERT #{insert.into} #{insert.values_list}"
+        sql = +"INSERT #{"IGNORE" if insert.skip_duplicates?} #{insert.into} #{insert.values_list}"
 
-        if insert.skip_duplicates?
-          no_op_column = quote_column_name(insert.keys.first)
-          sql << " ON DUPLICATE KEY UPDATE #{no_op_column}=#{no_op_column}"
-        elsif insert.update_duplicates?
```

Comment:
Before the change, if the following code is executed,
```
Book.insert_all [
  Book.new({ name: "Detective Stories", author_id: 1 }),
  Book.new({ name: "Detective Stories", author_id: 1 }),
]
```
The following sql is executed
```
"INSERT INTO `books` (`id`,`author_id`,`format`,`format_record_id`,`format_record_type`,`name`,`status`,`last_read`,`nullable_status`,`language`,`author_visibility`,`illustrator_visibility`,`font_size`,`difficulty`,`cover`,`isbn`,`external_id`,`original_name`,`published_on`,`boolean_status`,`tags_count`,`created_at`,`updated_at`,`updated_on`) VALUES (NULL, 1, NULL, NULL, NULL, 'Detective Stories', 0, 0, NULL, 0, 0, 0, 0, 0, 'hard', NULL, NULL, NULL, NULL, NULL, 0, CURRENT_TIMESTAMP(6), CURRENT_TIMESTAMP(6), CURRENT_TIMESTAMP(6)), (NULL, 1, NULL, NULL, NULL, 'Detective Stories', 0, 0, NULL, 0, 0, 0, 0, 0, 'hard', NULL, NULL, NULL, NULL, NULL, 0, CURRENT_TIMESTAMP(6), CURRENT_TIMESTAMP(6), CURRENT_TIMESTAMP(6)) ON DUPLICATE KEY UPDATE `id`=`id`"
```
and the following error will occur.
```
ActiveRecord::StatementInvalid: Mysql2::Error: Auto-increment value in UPDATE conflicts with internally generated values
```